### ISSUE TITLE
Pagination with $resources (or any $promise) does not work properly.

### ISSUE THREAD

**Issue by esseti** [angular-ui/bootstrap#3596]
Hello,
## problem:
I found that if I use the pagination setting as `total-items` a value that is retrieved from the api with `$resources` (then also via `$http`) it breaks the pagination (it always return to first page) since the `$watch` on the `totalItems` is fired twice.

## Ex:
I've a controller that performs a call to get the `courses`
```javascript
var fetchPage = function () {
  $scope.courses = gcResourceCourses.query({
    idClub: $stateParams.id,
    page: ($scope.currentPage - 1),
    size: $scope.itemsPerPage
  });
};

$scope.$watch('currentPage', fetchPage);
```

the api returns:
```json
{
  "total":23,
  "results":[...]
}
```

i set the pagination as this.
```html
<pagination class="pagination-sm"
  ng-model="currentPage"
  total-items="courses.total"
  boundary-links="false"
  items-per-page="itemsPerPage">
</pagination>
```

so if i click on a button of the pagination (let's say `3`) the `watch` is fired an the fetch page fetches page 3, now: the value is not yet set since the call is async. this means that the page is rendered with empty result, and then the `total-items` is set with a `null` value. This triggers the pagination `watch` on that value that set the `currentPage` to 1 resulting in having the pagination always set to 1.

## solution: 
(not sure it's correct but i guess so)
change the `pagination.js` [here](https://github.com/angular-ui/bootstrap/blob/master/src/pagination/pagination.js#L26) or [here](https://github.com/angular-ui/bootstrap/blob/master/src/pagination/pagination.js#L40) adding a control that checks that the value of `$scope.totalItems` is a number and not `undefined` (it should be `undefined` i guess)

*I'm posting this because i may know how to code but I'm not much into submitting contribution and I don't want to break things*

**Comment by wesleycho** [comment]
This is tricky - I believe there is another issue asking for the same feature.

The difficult part is that what should happen with the display if the total items value is missing? I think this needs some thought as to what is the best way to approach this.

**Comment by rainlake** [comment]
upvote, totalPages should not be set to 1 if totalItems is undefined.

**Comment by wesleycho** [comment]
This should be fixed by 30099a0cd080936b22027d37c52b3b32f1fb3ad7.
